client/blocks/inline-help/inline-help-search-results.jsx

```diff
@@ -21,92 +21,89 @@ import {
-	};
+function HelpSearchResults( {
+	hasAPIResults = false,
+	isSearching = false,
+	openResult,
+	searchQuery = '',
+	searchResults,
+	selectedResultIndex = -1,
+	selectSearchResult,
+	translate = identity,
+	placeholderLines = 4,
+} ) {
+	const selectResultHandler = useCallback( ( selectionIndex ) => ( event ) => {
+		const selectedResult = searchResults?.[ selectionIndex ] ?? null;
+		selectSearchResult( selectionIndex );
+		openResult( event, selectedResult );
+	} );
+
+	const renderHelpLink = ( { link, key, description, title }, index ) => {
+		const classes = classNames( 'inline-help__results-item', {
+			'is-selected': selectedResultIndex === index
+		} );
 
-	static defaultProps = {
-		translate: identity,
-		searchQuery: '',
+		return (
+			<li key={ link ?? key } className={ classes }>
+				<a
+					href={ localizeUrl( link ) }
+					onClick={ selectResultHandler( index ) }
+					title={ decodeEntities( description ) }
+				>
+					{ preventWidows( decodeEntities( title ) ) }
+				</a>
+			</li>
+		);
 	};
 
-	renderSearchResults() {
-		if ( this.props.isSearching ) {
+	const renderSearchResults = () => {
+		if ( isSearching ) {
 			// search, but no results so far
-			return <PlaceholderLines />;
+			return <PlaceholderLines lines={ placeholderLines } />;
 		}
 
-		// found something
-		const links = this.props.searchResults;
-
 		return (
 			<>
-				{ ! isEmpty( this.props.searchQuery ) && ! this.props.hasAPIResults && (
-					<p className="inline-help__empty-results">{ this.props.translate( 'No results.' ) }</p>
+				{ ! isEmpty( searchQuery ) && ! hasAPIResults && (
+					<p className="inline-help__empty-results">{ translate( 'No results.' ) }</p>
 				) }
-				<ul className="inline-help__results-list">{ links && links.map( this.renderHelpLink ) }</ul>
+				<ul className="inline-help__results-list">{
+					searchResults.map( renderHelpLink )
+				}</ul>
 			</>
 		);
-	}
-
-	onHelpLinkClick = ( selectionIndex ) => ( event ) => {
-		const selectedResult = this.props?.searchResults?.[ selectionIndex ] ?? null;
-		this.props.selectResult( selectionIndex );
-		this.props.openResult( event, selectedResult );
 	};
 
-	renderHelpLink = ( link, index ) => {
-		const classes = { 'is-selected': this.props.selectedResultIndex === index };
-		return (
-			<li
-				key={ link.link ? link.link : link.key }
-				className={ classNames( 'inline-help__results-item', classes ) }
-			>
-				<a
-					href={ localizeUrl( link.link ) }
-					onClick={ this.onHelpLinkClick( index ) }
-					title={ decodeEntities( link.description ) }
-				>
-					{ preventWidows( decodeEntities( link.title ) ) }
-				</a>
-			</li>
-		);
-	};
-
-	render() {
-		return (
-			<div>
-				<QueryInlineHelpSearch
-					query={ this.props.searchQuery }
-					requesting={ this.props.isSearching }
-				/>
-				{ this.renderSearchResults() }
-			</div>
-		);
-	}
+	return (
+		<div>
+			<QueryInlineHelpSearch query={ searchQuery } requesting={ isSearching } />
```

Comment:
This will reintroduce the issue whereby when the component renders we query HappyChat. This needs to be conditionally added based on whether the component is mounted in the FAB. 

I would make whatever prop you use to determine this agnostic of "being in the FAB". Rather make it about something generic as we don't want to tightly couple this component to the scenario of either "FAB" or "My Home".

Reply:
Ok I'm not so sure now. I might be completely wrong. Let's double check console for network requests like last time.

Reply:
I think the problem could be that both components mount when the app starts. We should handle this in a better way.

Reply:
I'm going to look into this a bit more as well.

Reply:
I don't see any search query API requests dispatched until I type a search.

I also don't see any HappyChat WebSocket connections as per the issue raised in https://github.com/Automattic/wp-calypso/pull/42993.

Therefore I'm going to say I was flagging something up that wasn't a problem. Apologies.